java-extras/src/test/java/org/triplea/java/RetryableTest.java

```diff
@@ -1,30 +1,43 @@
 package org.triplea.java;
 
+import static com.github.npathai.hamcrestopt.OptionalMatchers.isEmpty;
+import static com.github.npathai.hamcrestopt.OptionalMatchers.isPresent;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.Is.is;
 import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
 
 import java.time.Duration;
 import java.util.List;
-import java.util.function.BooleanSupplier;
+import java.util.Optional;
 import java.util.function.Consumer;
+import java.util.function.Supplier;
+import lombok.RequiredArgsConstructor;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.MethodSource;
 import org.junit.jupiter.params.provider.ValueSource;
-import org.mockito.Mock;
-import org.mockito.junit.jupiter.MockitoExtension;
 
-@ExtendWith(MockitoExtension.class)
 class RetryableTest {
 
-  @Mock private BooleanSupplier task;
-  @Mock private Consumer<Duration> threadSleeper;
+  /**
+   * Increment the invocation count and return true if the invocation count matches the input
+   * parameter. In other words, we can say "return true on the 3rd invocation."
+   */
+  @RequiredArgsConstructor
+  private static final class Task implements Supplier<Optional<Boolean>> {
+    private final int successIteration;
+    private int invocationCount;
+
+    @Override
+    public Optional<Boolean> get() {
+      invocationCount++;
+      return invocationCount == successIteration ? Optional.of(true) : Optional.empty();
+    }
+  }
+
+  private int sleepCount;
+  private final Consumer<Duration> threadSleeper = duration -> sleepCount++;
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6236397)